### ISSUE TITLE
Implement Gradient Accumulation for the training loop

### ISSUE THREAD

**Issue by xXCoolinXx** [xXCoolinXx/SPARBackdoor#4]
I currently haven't implemented gradient accumulation as it caused many weird errors when I tried. I believe I've solved the underlying issue, so that should now be possible.
